packages/do-not-merge/src/do-not-merge.ts

```diff
@@ -0,0 +1,126 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* eslint-disable node/no-extraneous-import */
+
+import {Application, Context} from 'probot';
+import Webhooks from '@octokit/webhooks';
+import {logger} from 'gcf-utils';
+
+const DO_NOT_MERGE = 'do not merge';
+const CHECK_NAME = 'Do Not Merge';
+
+const FAILURE_OUTPUT = {
+  title: 'Remove the do not merge label before merging',
+  summary: 'Remove the do not merge label before merging',
+};
+
+const SUCCESS_OUTPUT = {
+  title: 'OK to merge, label not found',
+  summary: 'OK to merge, label not found',
+};
+
+export = (app: Application) => {
+  app.on(
+    [
+      'pull_request.labeled',
+      'pull_request.unlabeled',
+      'pull_request.synchronize', // To run the check on every commit.
+    ],
+    async (
+      context: Context<Webhooks.EventPayloads.WebhookPayloadPullRequest>
+    ) => {
+      const owner = context.payload.repository.owner.login;
+      const repo = context.payload.repository.name;
+      const sha = context.payload.pull_request.head.sha;
+
+      const labelFound = context.payload.pull_request.labels.find(
+        l => l.name === DO_NOT_MERGE
+      );
+
+      const existingCheck = await findCheck(context, owner, repo, sha);
+
+      if (!labelFound) {
+        logger.info('Do not merge label not found');
+        // If the check already exists, but it's not a success, make it a success.
+        if (existingCheck && existingCheck.conclusion !== 'success') {
+          logger.info(
+            `Updating check on ${context.payload.pull_request.url} to success`
+          );
+          await context.github.checks.update({
```

Comment:
It feels like this will be super noisy, creating another status check on every PR in every repository across the org.  Would it possibly make more sense to only create the check if it's a failing PR?  That does of course come with the problem that there's no way to tell GitHub "don't allow merging of a PR of any check fails". 🤔

Reply:
This only updates to success when there is already an existing check, specifically for that reason. A check is only _added_ when it fails. However, we have to update the check to success if the label is removed. I don't think there is a way to dismiss a check entirely. See https://docs.github.com/en/free-pro-team@latest/rest/reference/checks#check-runs.